### PR TITLE
[FRAM-93] Add default options for normalization and denormalization

### PR DESCRIPTION
**Pull request by vincent4vx**

### CODE REVIEW

File: src/Serializer.php [hunk 1]
```diff
@@ -26,13 +26,31 @@ class Serializer implements SerializerInterface, NormalizerInterface, BinarySeri
...
     {
         $this->loader = $loader;
+
+        if ($defaultDenormalizationOptions !== null) {
+            $this->defaultDenormalizationContext = new DenormalizationContext($this, $defaultDenormalizationOptions);
+        }
+
+        if ($defaultNormalizationOptions !== null) {
+            $this->defaultNormalizationContext = new NormalizationContext($this, $defaultNormalizationOptions);
+        }
```
Comment (Johnmeurt) [inline src/Serializer.php:53]: This will cause a circular reference. We should store the context array and see what we can do on context instantiation

**Review by Johnmeurt — CHANGES_REQUESTED** [review]

File: src/Serializer.php [hunk 2]
```diff
@@ -208,4 +220,37 @@ public function getLoader(): NormalizerLoaderInterface
...
+     * @param array<string, mixed> $context
+     * @return DenormalizationContext
+     */
+    private function denormalizationContext(array $context): DenormalizationContext
+    {
+        if ($this->defaultDenormalizationOptions !== null) {
+            $context += $this->defaultDenormalizationOptions;
+        }
+
+        return new DenormalizationContext($this, $context);
```
Comment (Johnmeurt) [inline src/Serializer.php:236]: Maybe use the DenormalizationContext::duplicate() method as in the normalization context.

**Review by Johnmeurt — CHANGES_REQUESTED** [review]

File: src/Serializer.php (hunk 2 above)
Comment (vincent4vx) [reply src/Serializer.php:236]: This is not necessary, because there is no option alias. And the usage of duplicate has performance impacts.

Comment (Johnmeurt) [reply src/Serializer.php:236]: I know but this means the serializer knows how the context manage its options.

**Review by Johnmeurt — APPROVED** [review]

**Pull request merged**